contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c

```diff
@@ -2980,6 +2980,9 @@ TdsSendTypeNVarchar(FmgrInfo *finfo, Datum value, void *vMetaData)
 	}
 	else
 	{
+		/* We can store upto 2GB (2^31 - 1 bytes) for the nvarchar(max). */
+		if (unlikely(buf.len > VARCHAR_MAX))
+			elog(ERROR, "Number of bytes required for the field of nvarchar(max) exeeds 2GB");
```

Comment:
We should try not throw errors inside a sender

Reply:
Ideally, we should not be hitting this error in any case. So this error makes sense here (we are doing the same for varchar also).

Reply:
this is not even possible, int32 can have maximum value of VARCHAR_MAX, a larger value will mean len will become negative. There could be a sanity check for negative len though.